api/errors/error.go

```diff
@@ -0,0 +1,25 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker engine but are not tied to any specific top-level component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+const errGroup = "engine"
+
+var (
+	// ErrorCodeGeneric is generated when there is no better error message
+	// available. This will typically happen when some low-level function
+	// is called and we just need to bubble the unknown error back up to
+	// the API. We should try to avoid using this.
+	ErrorCodeGeneric = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
I decided to just remove this error definition for now. As I was changing to your suggested name I realized that I don't actually use it yet - I just added it with the assumption that I would need it as some point. But, let's wait until we need such a generic error message and worry about it then.